File: leetcode/python/1572_matrix_diagonal_sum.py
class Solution:
    def diagonalSum(self, mat: list[list[int]]) -> int:
        res = 0
        n = len(mat)
        for i in range(n):
            res += mat[i][i] + mat[i][n - 1 - i]
        if n % 2 == 1:
            res -= mat[n // 2][n // 2]
        return res

File: leetcode/rust/src/1572_matrix_diagonal_sum.rs
impl Solution {
    pub fn diagonal_sum(mat: Vec<Vec<i32>>) -> i32 {
        let mut res = 0;
        let n = mat.len();
        for i in 0..n {
            res += mat[i][i] + mat[i][n - i - 1];
        }
        if n % 2 == 1 {
            res - mat[(n / 2)][n / 2]
        } else {
            res
        }
    }
}